swag/swag_client.py

```diff
@@ -369,3 +416,294 @@ async def execute_swagdmin_command(self, message):
+            await message.channel.send(
+                "Transaction effectuée avec succès ! \n"
+                "```ini\n"
+                f"[{message.author.display_name}\t"
+                f"{format_number(value)} {currency_to_str(cagnotte_info.currency)}\t"
+                f"-->\t€{cagnotte_idx} {cagnotte_info.name}]\n"
+                "```"
+            )
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "donner" in splited_command:
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            cagnotte = self.swag_bank.get_active_cagnotte(cagnotte_idx)
+            destinataire = message.mentions
+            if len(destinataire) != 1:
+                await message.channel.send(
+                    "Merci de mentionner un destinataire (@Bobby Machin) pour "
+                    "lui donner une partie de cette €agnotte !"
+                )
+                return
+            destinataire = destinataire[0]
+
+            if cagnotte.currency == Currency.SWAG:
+                try:
+                    value = int(
+                        "".join(
+                            argent for argent in splited_command if argent.isnumeric()
+                        )
+                    )
+                except ValueError:
+                    raise InvalidSwagValue
+
+            elif cagnotte.currency == Currency.STYLE:
+                try:
+                    value = Decimal(
+                        "".join(
+                            argent
+                            for argent in splited_command
+                            if argent.replace(".", "").replace(",", "").isnumeric()
+                        )
+                    ).quantize(Decimal(".0001"), rounding=ROUND_DOWN)
+                except ValueError:
+                    raise InvalidStyleValue
+
+            self.swag_bank.receive_from_cagnotte(
+                cagnotte_idx, destinataire.id, value, message.author.id
+            )
+
+            cagnotte_info = cagnotte.get_info()
+            await message.channel.send(
+                "Transaction effectuée avec succès ! \n"
+                "```ini\n"
+                f"[€{cagnotte_idx} {cagnotte_info.name}\t"
+                f"{format_number(value)} {currency_to_str(cagnotte_info.currency)}\t"
+                f"-->\t{destinataire.display_name}]\n"
+                "```"
+            )
+
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "partager" in splited_command:
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            participants_id = [participant.id for participant in message.mentions]
+
+            (
+                participants_id,
+                gain,
+                gagnant_miette,
+                miette,
+            ) = self.swag_bank.share_cagnotte(
+                cagnotte_idx, participants_id, message.author.id
```

Comment:
`cagnotte_id`?

Reply:
idk why I prefer idx, because with idx (index) we know that's is something which begin by 0. Because of the discord_id which are big number, I get confused by the using of "id" in my code, so I prefer idx.

Reply:
But both are exactly the same thing; an integer identifier of an object in a table. It's just that, discord is billion times bigger than us and so they manage billions of objects instead of like two and a half xD

Reply:
You're right, but in my mind this difference is enough for me to say that they are "different". 

Making this distinction make my code workflow very efficent by removing stupid doubt, so if it's not a big deal I would like to keep it like that

Reply:
I'd like to argue the opposite. This distinction makes me stutter when I read it, because I don't see a difference between the two type of objects and my mind goes "uh, what is this again?". Also, since idx means index (or i when in a for loop for example), and an index is not the same thing as an identifier (even if, in this case, it is the same thing) also bugs me. To me, who works with scientific numerical array for work, it doesn't bear the meaning you feel it bears.

**HOWEVER** I *do* much much much agree we have to many id types, which are all `int`s, and that situation get pretty confusing and error prone. But I believe the solution is *typing* instead of *naming*.

***Therefore***, since we won't implement those in this pull request, and since the naming you chose is *not* part of public API (and even then, it whe can break our public API as much as we wan't since this is no library), I suggest we move on and close this discussion over the naming of a single variable. We can have that again, later.